patcher.sh

```diff
@@ -0,0 +1,339 @@
+#!/usr/bin/env bash
+
+# exit on any error to avoid showing everything was successfull even though it wasnt
+set -eou pipefail
+
+# Patching user variables
+OSCHOICE="linux"
+AM2RZIP=""
+HQMUSIC=false
+SYSTEMWIDE=false
+APPIMAGE=false
+PREFIX=""
+
+# Patching internal variables
+# Since people are likely to double click on this, I need a way to get the script_dir
+dirResult="$(dirname '${BASH_SOURCE[0]}')"
+SCRIPT_DIR="$(realpath $dirResult)"
+VERSION="1.5.5"
+GAMEDIR="${SCRIPT_DIR}/am2r_${VERSION}"
+RESOURCES="${GAMEDIR}/assets"
+
+patch_am2r ()
+{
+	# Set prefix to default value if empty
+	if [ -z "$PREFIX" ]; then
+		if [ "$SYSTEMWIDE" = true ] && [ ! "$OSCHOICE" = "android" ]; then
+			PREFIX="/usr/local"
+		else
+			PREFIX="$SCRIPT_DIR/am2r_${VERSION}"
+		fi
+	fi
+
+	# We need to do variable adjustments based on the prefix
+	if [ "$SYSTEMWIDE" = true ]; then
+		GAMEDIR="${PREFIX}/opt/am2r"
+		RESOURCES="${GAMEDIR}/assets"
+	else
+		GAMEDIR="${PREFIX}"
```

Comment:
This line is, to my understanding, substantially dangerous. $GAMEDIR gets rm -rf'd on line 43, and any directory name can be passed into this script as $PREFIX as per line 302. Say I'm a bit of a dimwitted user who works out of the desktop and decides to pass it in as the prefix - whole thing gets wiped just to dedicate the entire space to AM2R lol.

Reply:
Addressed in latest commit.